### PR TITLE
feat: support hardlink in --import-mode

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As follow up of #4620

Changes introduced in this pull request:

- add `--import-mode=hardlink` to hard link snapshots instead of copying them to save time and disk space
- add `--import-mode=auto` as default to hard link snapshots and fallback to copying them.
- add some info logs that print modes during snapshot import

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: scripts/tests/bootstrapper/docker-compose-forest.yml [hunk 1]
```diff
@@ -76,15 +76,16 @@ services:
...
+        data_dir = "/data/forest"
         [network]
         bootstrap_peers = ["/dns/forest-bootstrapper/tcp/$FOREST_P2P_PORT/p2p/$${PEER_ID}"]
         EOF
 
         forest --chain ${CHAIN} --encrypt-keystore false --no-gc \
           --config config.toml \
           --rpc-address 0.0.0.0:${FOREST_RPC_PORT} \
-          --import-snapshot $(ls /data/*.car.zst | tail -n 1) \
-          --import-mode=move
```
Comment (hanabi1224) [inline scripts/tests/bootstrapper/docker-compose-forest.yml:87]: It was previously copy & delete (which takes 13s in https://github.com/ChainSafe/forest/actions/runs/10783688527/job/29906201147#step:5:121)

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,6 +98,14 @@
   `Filecoin.EthGetTransactionByBlockNumberAndIndex` to existing methods (though
   without support, which matches the current Lotus's behavior).
 
+- [#4757](https://github.com/ChainSafe/forest/pull/4757) Added an option to
+  hardlink snapshots instead of moving or copying them. This can be invoked with
+  `--import-snapshot <path> --import-mode=hardlink`.
+
+- [#4757](https://github.com/ChainSafe/forest/pull/4757) Added an option to
+  hardlink snapshots and fallback to copying them if not applicable. This can be
+  invoked with `--import-snapshot <path> --import-mode=auto`.
```
Comment (LesnyRumcajs) [inline CHANGELOG.md:107]: This is a breaking change, no?

File: src/daemon/db_util.rs [hunk 3]
```diff
@@ -121,16 +126,62 @@ pub async fn import_chain_as_forest_car(
...
+                    from_path.display(),
+                    forest_car_db_path.display()
+                );
                 std::os::unix::fs::symlink(from_path, &forest_car_db_path)
                     .context("Error creating symlink")?;
             } else {
                 bail!("Snapshot file must be a valid forest.car.zst file");
             }
         }
+        ImportMode::Hardlink => {
```
Comment (LesnyRumcajs) [inline src/daemon/db_util.rs:172]: Can we add tests to different snapshot import modes in `src/test`?

File: CHANGELOG.md (hunk 2 above)
Comment (hanabi1224) [reply CHANGELOG.md]: Fixed. (Although user experience wise this change should be transparent to users)

File: src/daemon/db_util.rs (hunk 3 above)
Comment (hanabi1224) [reply src/daemon/db_util.rs:172]: Unit tests updated to cover `auto` and `hardlink` cases

Comment (LesnyRumcajs) [reply src/daemon/db_util.rs:172]: Lovely! I knew I created them somewhere but couldn't find them. :rofl:

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**